mimic/rest/maas_api.py

```diff
@@ -30,6 +30,16 @@
 Request.defaultContentType = 'application/json'
 
 
+def _find_index_of_id(xs, val):
```

Comment:
`list#index` compares elements by value, where this function only looks at the value of the `id` field. We could raise `IndexError` here instead of returning `-1`, which I like because `-1` is usually a valid index into a python list.